plugins/nf-prov/src/main/nextflow/prov/WrrocRenderer.groovy

```diff
@@ -49,213 +58,570 @@ class WrrocRenderer implements Renderer {
+                    "description"   : "",
+                    // TODO: apply only if type is Path
+                    // "encodingFormat": "text/plain",
+                    // TODO: match to output if type is Path
+                    // "workExample": ["@id": outputId],
+                    "name"          : name,
+                    // "valueRequired": "True"
+                ]
+            }
+
+        final inputFiles = workflowInputMapping
+            .collect { source, target ->
+                [
+                    "@id"           : crateRootDir.relativize(target).toString(),
+                    "@type"         : "File",
+                    "name"          : target.name,
+                    "description"   : "",
+                    "encodingFormat": Files.probeContentType(source) ?: "",
+                    // TODO: apply if matching param is found
+                    // "exampleOfWork": ["@id": paramId]
+                ]
+            }
 
         final outputFiles = workflowOutputs
-            .collect { source, target -> [
-                "@id": normalizePath(source),
-                "@type": "File",
-                "name": source.name,
-                "description": "",
-                "encodingFormat": Files.probeContentType(source) ?: "",
-                // TODO: create FormalParameter for each output file?
-                // "exampleOfWork": {"@id": "#reversed"}
-            ] }
+            .collect { source, target ->
+                [
+                    "@id"           : crateRootDir.relativize(target).toString(),
+                    "@type"         : "File",
+                    "name"          : target.name,
+                    "description"   : "",
+                    "encodingFormat": Files.probeContentType(target) ?: "",
+                    // TODO: create FormalParameter for each output file?
+                    // "exampleOfWork": {"@id": "#reversed"}
+                ]
+            }
+
+        // Combine both, inputFiles and outputFiles into one list. Remove duplicates that occur when an intermediate
+        // file is output of a task and input of another task.
+        Map<String, LinkedHashMap<String, String>> combinedInputOutputMap = [:]
+
+        inputFiles.each { entry ->
+            combinedInputOutputMap[entry['@id']] = entry
+        }
+        // Overwriting if 'id' already exists
+        outputFiles.each { entry ->
+            combinedInputOutputMap[entry['@id']] = entry
+        }
+        List<LinkedHashMap<String, String>> uniqueInputOutputFiles = combinedInputOutputMap.values().toList()
+
+        final propertyValues = params
+            .collect { name, value ->
+                [
+                    "@id"          : "#${name}-pv",
+                    "@type"        : "PropertyValue",
+                    "exampleOfWork": ["@id": "#${name}"],
+                    "name"         : name,
+                    "value"        : value
+                ]
+            }
```

Comment:
_During our Hackathon the following issue came up:_
Most nf-core pipelines deal with reference genome options (mainly igenomes). The way the #genome-pv section of the final ro-crate json is created leads to the whole rendered igenomes.config (example [here](https://github.com/nf-core/mag/blob/master/conf/igenomes.config)) to be added as value which makes "value" : {} nested and thus leads to an invalid crate. For our minimal example we just removed the reference genome option which then led to a non-nested crate again.

Our idea was to catch the genomes-pv case with an if statement, then copy the rendered igenomes.config to the results directory and point the crate value entry to that file.

Reply:
Thank you for identifying this problem. It didn’t occur during my tests with the demo pipeline. Catching the `genomes-pv` case might help for nf-core workflows that use iGenomes, but it doesn’t address the underlying issue. Off the top of my head, I’d suggest putting the entire content into one big string and storing it as a `value`. I’ll try that this afternoon and check if the validator accepts it.

Reply:
We removed the igenomes reference from our small test pipeline for now. But there has to be a more elegant way :) I also treid some stuff on top of what you did and expanded the ro-crate (see https://github.com/nextflow-io/nf-prov/pull/37).

Reply:
I ran rnaseq 3.14.0 on our cluster (where we have igenomes) and validated `ro-crate-metadata.json` with the newest validator 0.4.5. The validator does not complain about the `value` in the `#genome-pv` section.
I pasted the section in a JSON validator and it's also valid JSON.
Can you upload your problematic RO-Crate somewhere and tell me which rocrate-validator version you used?

Reply:
@mburridge96 I think you have the better overview about the issue with the nested crates. Maybe you can explain :)

Reply:
Whilst it's valid JSON, the RO-Crate spec requires every object within the @Graph array to be flat. So every object that's in the array must have an "@id", "@type" and then any additional key:value pairs. If nested data is required, you then point to another object that describes it.

So below is valid:
```json
"@graph" : [{
"@id" : "URI/#unique_local",
"@type": "schema_type", 
"name":  "example",
"description": "description of example",
"about": {"@id": "linktoabout"},
}, {
"@id": "linktoabout",
"@type": "schema_type"
}]
```

but then the following is not:
```json
"@graph" : [{
"@id" : "URI/#unique_local",
"@type": "schema_type", 
"name":  "example",
"description": "description of example",
"about": {
"@id": "linktoabout",
"@type": "schema_type"
}}]
```

The issue is if we dump everything in the value key it won't conform to the RO-Crate spec and I don't think (could be wrong) that the validator yet checks for that yet which is another problem in itself.

Reply:
@famosab & @mburridge96 I think I fixed this in 4d22b8425a0f6c35f196f7d7dd8a71ec5946bd84.
I'm checking if the `value` is a Groovy List or Map and save it as String instead of nested JSON.
It works with the latest validator.